sharded_queue/api.lua

```diff
@@ -402,8 +422,142 @@ function sharded_tube.drop(self)
 end
 
 local sharded_queue = {
-    tube = {}
+    tube = {},
+    cfg = stash_get(stash_names.cfg),
+    metrics_stats = stash_get(stash_names.metrics_stats),
 }
+if sharded_queue.cfg.metrics == nil then
+    sharded_queue.cfg.metrics = true
+end
+
+local function is_metrics_v_0_11_installed()
+    if not is_metrics_package or metrics.unregister_callback == nil then
+        return false
+    end
+    local counter = require('metrics.collectors.counter')
+    return counter.remove and true or false
+end
+
+local function metrics_create_collectors()
+    local create_counter = function(name, description)
+        return {
+            collector = metrics.counter(name, description),
+            values = {},
+        }
+    end
+    local create_gauge = function(name, description)
+        return {
+            collector = metrics.gauge(name, description)
+        }
+    end
+    local call_names = {
+        "done",
+        "take",
+        "kick",
+        "bury",
+        "put",
+        "delete",
+        "touch",
+        "ack",
+        "release",
+    }
+
+    local calls = {}
+    for _, name in ipairs(call_names) do
+        calls[name] = create_counter(
+            "sharded_queue_calls_" .. name,
```

Comment:
It sounds like it could be a label instead of name suffix.

Reply:
It is common to have `_total` suffix for accumulated counter values in Prometheus.

Reply:
`_all` ?

Reply:
Do you mean to use the collector like this?
```
collector:inc(inc, {tube = tube_name, call = call_name})
```

Reply:
So, at now there are two collectors:

1. `sharded_queue_calls` a counter with labels `{name = call_name, path = task_name}`
2. `sharded_queue_tasks` a gauge with labels `{name = tasks_name, path = task_name}`

Reply:
How's about `{ name = task_name, status = call_name/tasks_name}`? `{ name = 'my_tube', status = 'done'}` seems less confusing than `{ path = 'my_tube', name = 'done'}`

Reply:
Of course, no problem - I rely on your experience with `metrics`. Updated.